### ISSUE TITLE
Add yarn berry support

### ISSUE THREAD

**Issue by wdfinch** [vercel/turborepo#237]
### Describe the feature you'd like to request

I would like to be able to use this tool with [yarn berry](https://github.com/yarnpkg/berry) as my package manager. On the docs it says only yarn v1 is [supported right now.](https://turborepo.org/docs/getting-started#add-turborepo-to-your-existing-monorepo)

### Describe the solution you'd like

Support for [yarn berry](https://github.com/yarnpkg/berry). 

### Describe alternatives you've considered

Using another package manager.

**Comment by tadjohnston** [comment]
@wdfinch I'm assuming you want pnp mode? It works as is with yarn berry using `nodeLinker: node-modules`, just not pnp.

**Comment by TrySound** [comment]
Is it related to only pnp support or whole yarn 2+?

**Comment by jaredpalmer** [comment]
It likely doesn't work as expected with `nodeLinker: node-modules` fyi. Need to test this out

**Comment by tadjohnston** [comment]
> It likely doesn't work as expected with `nodeLinker: node-modules` fyi. Need to test this out

Weird. I swear I had this working just fine (at least it was building fine) with `node-modules` set. I'll test again.

**Comment by tundera** [comment]
> It likely doesn't work as expected with `nodeLinker: node-modules` fyi. Need to test this out

I tried it on a design system codebase set up with yarn v3 using `nodeLinker: node-modules`. Everything in my pipeline is working, but I'm wondering if I'll run into an edge case soon. Most likely going to have a branch on v3 and revert to v1 on `main` until some of the edge cases are identified. Switching to v3 is a big boost in speed on installs, and I'm not even using the `pnp` linker.

**Comment by Jakst** [comment]
> > It likely doesn't work as expected with `nodeLinker: node-modules` fyi. Need to test this out
> 
> I tried it on a design system codebase set up with yarn v3 using `nodeLinker: node-modules`. Everything in my pipeline is working, but I'm wondering if I'll run into an edge case soon. Most likely going to have a branch on v3 and revert to v1 on `main` until some of the edge cases are identified. Switching to v3 is a big boost in speed on installs, and I'm not even using the `pnp` linker.

Lockfile marshalling does not yet work for yarn v2/v3. See https://github.com/vercel/turborepo/issues/268

**Comment by tundera** [comment]
@Jakst Oh nice catch! Back to v1 it is

**Comment by subgan82** [comment]
We are using `yarn 2 ` with nodeLinker: node-modules, seeing that it does not hit the cache for packages that don't have any code changes. Is this also related to #268 ?

**Comment by chentsulin** [comment]
Is there any plan or timeline for node_modules linker and PnP support?
I guess many people are waiting on this, and they will eventually find this thread. Any additional information is helpful. 😄

**Comment by harrytran998** [comment]
. for a vote this feature!

**Comment by wdfinch** [comment]
@jaredpalmer thanks for linking the PR. From the PR it appears that this only provides support when the node_modules linker is used. Given this ticket requests support for node_modules linker and PnP, I believe we should leave this issue open until PnP support is added.

I can also make a new ticket if you would prefer to break it up.

**Comment by trivikr** [comment]
> I believe we should leave this issue open until PnP support is added.
> I can also make a new ticket if you would prefer to break it up.

There haven't been any updates on this issue for three weeks.
@wdfinch Can you create a new issue for tracking?

**Comment by wdfinch** [comment]
@trivikr I made a new ticket that is focused on PnP support. I changed the title here to be on adding node_modules linker support since that's what the linked PR does.

Please move the discussion around adding PnP support to #693.

Additionally, yarn berry v2 is deprecated so we should only be adding support for v3. I'm not sure if that matters regarding the fix, but hopefully it limits the scope.
